app/views/exception_notifier/_session.text.erb

```diff
@@ -0,0 +1,20 @@
+<% browser = Browser.new(@request.user_agent) %>
+Browser Name: <%= browser.name %>
+Browser Version: <%= browser.full_version %>
+Browser Platform: <%= browser.platform %> <%= browser.platform.version %>
+Browser Device: <%= browser.device.name %>
+Browser Bot?: <%= browser.bot? %>
+
+Referer: <%= @request.referer %>
+
+<% session = @request.session.to_hash %>
+<% if session['sp'] %>
+  <% session['sp'] = session['sp'].except('request_url') %>
+  <% session['sp']['request_id'] = 'FILTERED' if session['sp']['request_id'] %>
+<% end %>
+<% session['user_return_to'] = session['user_return_to']&.split('?')&.first %>
+Session: <%= session.except('warden.user.user.key', 'warden.user.user.session', '_csrf_token') %>
```

Comment:
This makes me thing we may want to add a checkbox to our PR checklist to make sure that when you are adding user data to the session, you use the `user_session` helper and not the `session` helper.

Reply:
Opened here: https://github.com/18F/identity-idp/pull/1796

Reply:
Good idea. The `warden*` and `user_return_to` keys mentioned here all come from external gems: Devise and Warden. Also, some data, like the SP-related info, cannot be added to `user_session` because we don't know who the user is yet until they sign in.

Reply:
Alternatively, since I'm only interested in a few specific keys here, like the Service Provider and `user_return_to`, I will only ask for those keys and ignore everything else.

Reply:
I do get a little bit nervous about this because it seems too easy for someone to add something to the session that we don't want showing up here, and forgetting about this little piece of code. What are your thoughts on using a whitelist here instead of a blacklist?

Reply:
☝️ @monfresh You're a faster typist than me.

Reply:
Looks like we would have been fine by excluding `warden.user.user.session` because that's where everything we store in the `user_session` goes, but I went ahead and took the whitelist approach for safety and clarity. Please take another look.

Reply:
thanks. This looks a lot safer to me.